pkg/controller/parent/controller.go

```diff
@@ -110,18 +110,38 @@ func (r *ParentReconciler) Reconcile(ctx context.Context, req reconcile.Request)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+	resourceControllerConfig := resourceconfig.LoadConfig()
+	if !resourceControllerConfig.IsControllerSupported(r.gvk, controllerType) {
+		logger.Info("controller type not supported for this resource", "resource", req.NamespacedName, "type", controllerType)
+		config, err := resourceControllerConfig.GetControllersForGVK(r.gvk)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("error getting controller config found for GroupKind %v", r.gvk.GroupKind())
+		}
+		logger.Info("supported controller types", "resource", req.NamespacedName, "supportedControllers", config.SupportedControllers)
+		controllerType = config.DefaultController
```

Comment:
A little worried this will tend to "hide" the problem. Can we put an error on the CCC status?